src/modules/colorPicker/ColorPickerUI/Mouse/CursorManager.cs

```diff
@@ -0,0 +1,73 @@
+﻿using ColorPicker.Helpers;
+using Microsoft.Win32;
+using System;
+using System.IO;
+
+namespace ColorPicker.Mouse
+{
+    public static class CursorManager
+    {
+        private const string CursorsRegistryPath = @"HKEY_CURRENT_USER\Control Panel\Cursors\";
+        private const string ArrowRegistryName = "Arrow";
+        private const string IBeamRegistryName = "IBeam";
+        private const string CrosshairRegistryName = "Crosshair";
+        private const string HandRegistryName = "Hand";
+        private const string ColorPickerCursorName = "Resources\\colorPicker.cur";
+
+        private static string _originalArrowCursorPath;
+        private static string _originalIBeamCursorPath;
+        private static string _originalCrosshairCursorPath;
+        private static string _originalHandCursorPath;
+
+        public static void SetColorPickerCursor()
+        {
+            BackupOriginalCursors();
+
+            var colorPickerCursorPath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, ColorPickerCursorName);
+            ChangeCursor(colorPickerCursorPath, ArrowRegistryName);
+            ChangeCursor(colorPickerCursorPath, IBeamRegistryName);
+        }
+
+        public static void RestoreOriginalCursors()
+        {
+            ChangeCursor(_originalArrowCursorPath, ArrowRegistryName);
+            ChangeCursor(_originalIBeamCursorPath, IBeamRegistryName);
+        }
+
+        private static void ChangeCursor(string curFile, string cursorRegistryName)
+        {
+            try
+            {
+                Registry.SetValue(CursorsRegistryPath, cursorRegistryName, curFile);
+                Win32Apis.SystemParametersInfo(SPI_SETCURSORS, 0, new IntPtr(0), SPIF_SENDCHANGE);
```

Comment:
Can this be done with  an API or potentially P/Invoking [SetCursor](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-setcursor) instead of modifying the registry?

Reply:
Unfortunately no. SetCursor cannot change the actual "image" used for the cursor, it can only switch between the ones that Windows knows about, therefore we have change registry first.
For more discussion about this:
https://stackoverflow.com/a/15016483/613299

Reply:
@martinchrzan Gotcha. I think we need to be be very careful when terminating the process that we aren't putting peoples machines in a state where the only way to fix it would be to modify the registry.

Reply:
Yeah, this whole changing cursor is a little tricky but I used such a cursor which you can still use for a while if process dies unexpectedly.
If this happens, you don't have to change registry - you can go to mouse settings and do any cursor change such as - change cursor size or a set cursor - it will restore back to the original one - Windows is keeping a copy of that setting somewhere so it will not be destroyed.
![image](https://user-images.githubusercontent.com/11967522/87464190-ff1ca600-c612-11ea-981f-a0c0fd56a677.png)

That said, there is no real "save" way of changing cursor. I have added the option in the setting where you can turn off changing the cursor if it gives you any problems or you just don't like it.